_chapters/functionaljavascript.md

```diff
@@ -290,16 +290,129 @@ For many standard loops, however, the logic is the same every time and can easil
     const result = someArray.reduce((acc, el) => el * acc, 1);
     ```
 
+---
+
+#### Exercises
+
+1. Refactor this code to use [`map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) and [`filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) instead of a loop:
+
+    ```javascript
+    const numbers = [2, 6, 3, 7, 10];
+    const result = [];
+    for (let i = 0; i < numbers.length; i++) {
+        if (numbers[i] % 2 === 0) {
+            result.push(numbers[i] / 2);
+        }
+    }
+    ```
+
+2. Refactor this code to remove the loop:
+
+    ```javascript
+    const words = ["apple", "banana", "cherry"];
+    let totalLength = 0;
+    for (let i = 0; i < words.length; i++) {
+        totalLength += words[i].length;
+    }
+    ```
+
+3. Refactor this code to remove the loop:
+
+    ```javascript
+    const people = [
+        {name: "Alice", age: 20},
+        {name: "Bob", age: 15},
+        {name: "Charlie", age: 30},
+        {name: "David", age: 10},
+    ];
+    let firstChildName = undefined;
+    for (let i = 0; i < people.length; i++) {
+        if (people[i].age < 18) {
+            firstChildName = people[i].name;
+            break;
+        }
+    }
+    ```
+
+4. Refactor this code to remove the loop:
+
+    ```javascript
+    const people = [
+        {name: "Alice", age: 20},
+        {name: "Bob", age: 15},
+        {name: "Charlie", age: 30},
+        {name: "David", age: 10},
+    ];
+    let result = "";
+    for (let i = 0; i < people.length; i++) {
+        result += `Person #${i + 1}: ${people[i].name} is ${people[i].age} years old`;
+        if (i != people.length - 1) {
+            result += "\n";
+        }
+    }
+    ```
+
+#### Solutions
+
+1. We can use `filter` to get the even numbers, then `map` to divide each of those numbers by 2:
+
+    ```javascript
+    const numbers = [2, 6, 3, 7, 10];
+    const result = numbers.filter(x => x % 2 === 0).map(x => x / 2);
+    ```
+
+2. We need to sum up the lengths of each word. One way to do that would be to use `reduce`:
+
+    ```javascript
+    const words = ["apple", "banana", "cherry"];
+    const totalLength = words.reduce((acc, x) => acc + x.length);
```

Comment:
please comment on accumulator and why you can it be missing

Reply:
ignore my grammar i got distracted by shiny object half way through the sentence

Reply:
oops, fixed